pkg/apiclient/argo-kube-client.go

```diff
@@ -84,13 +86,16 @@ func newArgoKubeClient(ctx context.Context, clientConfig clientcmd.ClientConfig,
 	if err != nil {
 		return nil, nil, err
 	}
-	return ctx, &argoKubeClient{instanceIDService}, nil
+	wfStore, err := store.NewSQLiteStore(instanceIDService)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
in particular, this looks potentially off to me, the _client_ shouldn't need a store, only the Server should need that

Reply:
@jiachengxu confirmed this [on Slack](https://cloud-native.slack.com/archives/C0510EUH90V/p1715189885251799?thread_ts=1714810708.644859&cid=C0510EUH90V). We're still figuring out the appropriate fix for this though

Reply:
I managed to fix the issue with a bit refactoring and abstraction of the code, this commit fixes the issue: https://github.com/argoproj/argo-workflows/pull/13021/commits/14a11367a653d7c312b490f0e0526d8e481e250f
You are right about the issue, in the fix, the sqlite store is only used in server mode and not in cli mode, for cli mode, we still talk to kubernetes to get the live workflows.

Reply:
> for cli mode, we still talk to kubernetes to get the live workflows.

Or the Argo Server instead of k8s in the CLI's `HTTP` mode

Reply:
Resolved in #13021